File: deprecated/examples/example7.pl
#!/usr/bin/perl
#
# File: example7.pl       I. Henry   March 30 2005
#
# Perl translation of example7.c from the WFDB Programmer's Guide
# 							   
# Copyright (C) 2005 Isaac C. Henry (ihenry@physionet.org)

package wfdb;
use wfdb;

$nc = @ARGV - 3;
if ( @ARGV < 3 ) {
    print STDERR "usage: ", $0, " record start duration [ coefficients ...]\n";
    exit(1);
} 
if ($nc < 1) {
    $nc = 1;
    $c = 1.0;
} 
for ($i = 0; $i < $nc; $i++) {
    $c[$i] = $ARGV[$i+3];
}
exit(3) if (($nsig = isigopen($ARGV[0], undef, 0)) < 1);
$s = new wfdb::WFDB_SiginfoArray($nsig);
$v = new wfdb::WFDB_SampleArray($nsig);
exit(3) if (isigopen($ARGV[0], $s->cast(), $nsig) != $nsig);
if (isigsettime(strtim($ARGV[1])) < 0) {
    print strtim($ARGV[1]), "\n";
    exit(4);
}
if (($nsamp = strtim($ARGV[2])) < 1) {
    print STDERR $0, ": inappropriate value for duration\n";
    exit(5);
}
exit(6) if (osigopen("16l", $s->cast(), $nsig) != $nsig);
sample(0, 0);
for ($t = 0; $t < $nsamp && sample_valid(); $t++) {
    for ($j = 0; $j < $nsig; $j++) {
        for ($i = 0, $vv = 0.; $i < $nc; $i++) {
            $vv += $c[$i]*sample($j, $t+$i) if ($c[$i] != 0.);
	}
        $v->setitem($j, int($vv));
    }
    last if (putvec($v->cast()) < 0);    
}
newheader("out");
wfdbquit();

File: deprecated/examples/example7.cs
// File: example7.cs         I. Henry    February 18 2005
//
// C# translation of example7.c from the WFDB Programmer's Guide
//
// Copyright (C) 2005 Isaac C. Henry (ihenry@physionet.org)

using System;
using Wfdb;

public class example7 {
    static void Main(string[] argv) {
	double[] c; double vv;
	int i, j, nc = argv.Length - 4, nsig, nsamp, t;
		
	if (argv.Length < 3) {
	    Console.WriteLine(
		 "usage: example7 record start duration [ coefficients ... ]");
	    // Unlike C programs, C# programs do not have any foolproof way
	    // to discover their own names, so the name is given as a constant
	    // above.  The command needed to run this program within a VM
	    // is platform-dependent and likely to be more complex.
	    Environment.Exit(1);
	}
	if (nc < 1) {
	    nc = 1;
	    c = new double[1];
	    c[0] = 1.0;
	}
	else {
	    c = new double[nc];
	}
	for (i = 0; i < nc; i++)
	    c[i] = Double.Parse(argv[i+3]);
	if ((nsig = wfdb.isigopen(argv[0], null, 0)) < 1)
	    Environment.Exit(3);
	WFDB_SiginfoArray s = new WFDB_SiginfoArray(nsig);
	WFDB_SampleArray v = new WFDB_SampleArray(nsig);
	if (wfdb.isigopen(argv[0], s.cast(), nsig) != nsig)
	    Environment.Exit(3);
	if (wfdb.isigsettime(wfdb.strtim(argv[1])) < 0)
	    Environment.Exit(4);
	if ((nsamp = wfdb.strtim(argv[2])) < 1) {
	    Console.WriteLine("example7: inappropriate value for duration");
	    Environment.Exit(5);
	}
	if (wfdb.osigopen("16l", s.cast(), (uint)nsig) != nsig)
	    Environment.Exit(6);
		
	wfdb.sample(0, 0);
	for (t = 0; t < nsamp && wfdb.sample_valid() == 1; t++) {
	    for (j = 0; j < nsig; j++) {
		for (i = 0, vv = 0.0; i < nc; i++)
		    if (c[i] != 0.0) vv += c[i]*wfdb.sample((uint)j, t+i);
		v.setitem(j, (int)vv);
	    }
	    if (wfdb.putvec(v.cast()) < 0) break;
	}
		
	wfdb.newheader("out");
	wfdb.wfdbquit();
    }
}
